src/platform/forms-system/src/js/review/ReviewCollapsibleChapter.jsx

```diff
@@ -83,208 +83,253 @@ class ReviewCollapsibleChapter extends React.Component {
+    let chapterTitle = chapterFormConfig.title;
+    if (typeof chapterFormConfig.title === 'function') {
+      chapterTitle = chapterFormConfig.title(true);
+    }
+    if (chapterFormConfig.reviewTitle) {
+      chapterTitle = chapterFormConfig.reviewTitle;
+    }
+    return chapterTitle;
+  };
 
+  getSchemaformPageContent = (page, props) => {
     const {
       chapterFormConfig,
       expandedPages,
       form,
       formContext,
-      pageKeys,
-      hasUnviewedPages,
       viewedPages,
-    } = this.props;
+    } = props;
 
-    const ChapterDescription = chapterFormConfig.reviewDescription;
-    let chapterTitle = chapterFormConfig.title;
-    if (typeof chapterFormConfig.title === 'function') {
-      chapterTitle = chapterFormConfig.title(true);
+    const pageState = form.pages[page.pageKey];
+    let pageSchema;
+    let pageUiSchema;
+    let pageData;
+    let arrayFields;
+    let editing;
+    let fullPageKey;
+
+    if (page.showPagePerItem) {
+      editing = pageState.editMode[page.index];
+      pageSchema =
+        pageState.schema.properties[page.arrayPath].items[page.index];
+      pageUiSchema = pageState.uiSchema[page.arrayPath].items;
+      pageData = _.get([page.arrayPath, page.index], form.data);
+      arrayFields = [];
+      fullPageKey = `${page.pageKey}${page.index}`;
+    } else {
+      editing = pageState.editMode;
+      // TODO: support array fields inside of an array page?
+      // Our pattern is to separate out array fields (growable tables) from
+      // the normal page and display them separately. The review version of
+      // ObjectField will hide them in the main section.
+      arrayFields = getArrayFields(pageState, page);
+      // This will be undefined if there are no fields other than an array
+      // in a page, in which case we won’t render the form, just the array
+      const pageSchemaObjects = getNonArraySchema(
+        pageState.schema,
+        pageState.uiSchema,
+      );
+      pageSchema = pageSchemaObjects.schema;
+      pageUiSchema = pageSchemaObjects.uiSchema;
+      pageData = form.data;
+      fullPageKey = page.pageKey;
     }
-    if (chapterFormConfig.reviewTitle) {
-      chapterTitle = chapterFormConfig.reviewTitle;
+
+    const classes = classNames('form-review-panel-page', {
+      'schemaform-review-page-warning': !viewedPages.has(fullPageKey),
+      // Remove bottom margin when the div content is empty
+      'vads-u-margin-bottom--0': !pageSchema && arrayFields.length === 0,
+    });
+    const title = page.reviewTitle || page.title || '';
+    const ariaLabel = `Update ${(typeof title === 'function'
+      ? title(pageData)
+      : title) || 'page'}`;
+
+    const visibleFields =
+      pageSchema &&
+      Object.entries(pageSchema.properties).filter(([propName]) =>
+        showReviewField(
+          propName,
+          pageSchema,
+          pageUiSchema,
+          form.data,
+          formContext,
+        ),
+      ).length > 0;
```

Comment:
This changed from the original. Previously, it was `noVisibleFields`. I changed to `visibleFields` because:
- It's easier to reason about.
- When `pageSchema` is `undefined`, such as when we've got an empty `schema.properties` because we have a `CustomPage`, the previous statement (`pageSchema && !Object.entries(...).filter(...).length > 0`) was evaluating to false—so `!noVisibleFields` because `pageSchema` doesn't exist. I needed it to mean "there are no visible fields when there's no schema," so I refactored and fixed that little bug.